src/synology_dsm/api/core/external_usb.py

```diff
@@ -0,0 +1,204 @@
+            if device["dev_id"] == device_id:
+                return device
+        return {}
+
+    def device_name(self, device_id):
+        """The title of theexternal USB storage device."""
+        return self.get_device(device_id).get("dev_title")
+
+    def device_type(self, device_id):
+        """The type of theexternal USB storage device."""
+        return self.get_device(device_id).get("dev_type")
+
+    def device_size_total(self, device_id, human_readable=False):
+        """Total size of theexternal USB storage device."""
+        return_data = int(self.get_device(device_id).get("total_size_mb"))
+        return_data = SynoFormatHelper.megabytes_to_bytes(return_data)
+        if human_readable:
+            return SynoFormatHelper.bytes_to_readable(return_data)
+        return return_data
+
+    def device_status(self, device_id):
+        """The status of the external USB storage device."""
+        return self.get_device(device_id).get("status")
+
+    def device_formatable(self, device_id):
+        """Whether the external USB storage device can be formatted."""
+        return self.get_device(device_id).get("formatable")
+
+    def device_progress(self, device_id):
+        """The progress the external USB storage device."""
+        return self.get_device(device_id).get("progress")
+
+    def product_name(self, device_id):
+        """The product name of the external USB storage device."""
+        return self.get_device(device_id).get("product")
+
+    def producer(self, device_id):
+        """The producer name of the external USB storage device."""
+        return self.get_device(device_id).get("producer")
+
+    # Partition
+    def device_partitions(self, device_id) -> list[SynoUSBStoragePartition]:
+        """Returns all partitions of the external USB storage device."""
+        partitions = self.get_device(device_id).get("partitions", [])
+        device_partitions: list[SynoUSBStoragePartition] = []
+        for partition in partitions:
+            device_partitions.append(SynoUSBStoragePartition(partition))
+        return device_partitions
+
+    def device_partition_ids(self, device_id):
+        """Returns partition ids of the external USB storage device."""
+        partitions = self.get_device(device_id).get("partitions", [])
+        partition_ids = []
+        for partition in partitions:
+            partition_ids.append(partition["name_id"])
+        return partition_ids
+
+    def device_partition(self, device_id, partition_id):
+        """Returns a partition of the external USB storage device."""
+        for partition in self.device_partitions(device_id):
+            if partition.name_id == partition_id:
+                return partition
+        return None
+
+    def partitions_all_size_total(self, device_id, human_readable=False):
+        """Total size of all parititions of the external USB storage device."""
+        partitions = self.device_partitions(device_id)
+        if partitions:
```

Comment:
just return None if there are no partitions. this will save one level of indentation

Reply:
I don't understand what you mean, I think I'm not familiar enough with Python yet.